backend/src/main/java/org/globe42/dao/PostalCityDaoImpl.java

```diff
@@ -0,0 +1,81 @@
+package org.globe42.dao;
+
+import java.text.Normalizer;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.TypedQuery;
+
+import org.globe42.domain.PostalCity;
+
+/**
+ * Implementation of custop methods of {@link PostalCityDao}
+ * @author JB Nizet
+ */
+public class PostalCityDaoImpl implements PostalCityDaoCustom {
+
+    @PersistenceContext
+    private EntityManager em;
+
+    @Override
+    public void saveAllEfficiently(Collection<PostalCity> cities) {
+        int counter = 0;
+        for (PostalCity city : cities) {
+            em.persist(city);
+            counter++;
+            if (counter == 50) {
```

Comment:
Yes, it's still one transaction. I picked 50 a bit arbitrarily. We probably would gain a little bit much in this usecase (but not much, I vaguely measured) by activating batching of inserts, but the main advantage of this technique is to avoid filling the first-level cache (and thus the memory) with lots of entities that are not needed anymore once inserted, and that take a lot of time to dirty-check before flush. I didn't really measure the different compared to not doing that, because I know, from experience and from the doc, that it's very slow to not do it.